modules/storage/cosmos-db/main.bicep

```diff
@@ -1,99 +1,353 @@
+  ipAddressOrRange: ipRule
+}]
+
+var varPrivateEndpoints = [for endpoint in privateEndpoints: {
+  name: '${cosmosDBAccount.name}-${endpoint.name}'
+  privateLinkServiceId: cosmosDBAccount.id
+  groupIds: [
+    endpoint.groupId
+  ]
+  subnetId: endpoint.subnetId
+  privateDnsZones: contains(endpoint, 'privateDnsZoneId') ? [
+    {
+      name: 'default'
+      zoneId: endpoint.privateDnsZoneId
+    }
+  ] : []
+  manualApprovalEnabled: contains(endpoint, 'manualApprovalEnabled') ? endpoint.manualApprovalEnabled : false
+}]
+
+resource cosmosDBAccount 'Microsoft.DocumentDB/databaseAccounts@2022-11-15' = {
   name: toLower(name)
   location: location
-  kind: 'GlobalDocumentDB'
+  kind: (backendApi == 'mongodb') ? 'MongoDB' : 'GlobalDocumentDB'
   properties: {
-    consistencyPolicy: consistencyPolicy[defaultConsistencyLevel]
-    locations: locations
+    analyticalStorageConfiguration: enableAnalyticalStorage ? { schemaType: analyticalStorageSchemaType } : null
+    apiProperties: (backendApi == 'mongodb') ? { serverVersion: serverVersion } : null
+
+    capabilities: [for capability in varCapabilities: { name: capability }]
+    capacity: enableServerless ? null : { totalThroughputLimit: totalThroughputLimit }
+    consistencyPolicy: varConsistencyPolicy
+    cors: cors
     databaseAccountOfferType: 'Standard'
-    enableAutomaticFailover: systemManagedFailover
-    enableMultipleWriteLocations: enableMultipleWriteLocations
-    capabilities: capabilities
+    disableKeyBasedMetadataWriteAccess: disableKeyBasedMetadataWriteAccess
+    disableLocalAuth: disableLocalAuth
+    enableAnalyticalStorage: enableAnalyticalStorage
+    enableAutomaticFailover: enableAutomaticFailover
+    enableFreeTier: enableFreeTier
+    enableMultipleWriteLocations: enableServerless ? false : enableMultipleWriteLocations
+    ipRules: varIpRules
+    isVirtualNetworkFilterEnabled: isVirtualNetworkFilterEnabled
+    locations: varLocations
+    networkAclBypass: networkAclBypass
+    networkAclBypassResourceIds: networkAclBypassResourceIds
+    publicNetworkAccess: publicNetworkAccess
+    virtualNetworkRules: virtualNetworkRules
   }
+  identity: contains(identityType, 'UserAssigned') ? {
+    type: identityType
+    userAssignedIdentities: contains(identityType, 'UserAssigned') ? userAssignedIdentities : {}
+  } : { type: identityType }
+  tags: tags
 }
 
-@description('Cosmos DB Resource ID')
-output id string = cosmosDB.id
+@batchSize(1)
+module cosmosDBAccount_cassandraKeyspaces 'modules/cassandra.bicep' = [for (keyspace, index) in cassandraKeyspaces: if (backendApi == 'cassandra') {
+  name: 'cassandra-keyspace-${uniqueString(keyspace.name, resourceGroup().name)}-${index}'
+  dependsOn: [
+    cosmosDBAccount
+  ]
+  params: {
+    cosmosDBAccountName: name
+    keyspaceName: keyspace.name
+    cassandraTables: keyspace.tables
+    autoscaleMaxThroughput: contains(keyspace, 'autoscaleMaxThroughput') ? keyspace.autoscaleMaxThroughput : 0
```

Comment:
```suggestion
    autoscaleMaxThroughput: keyspace.autoscaleMaxThroughput ?? 0
```

Reply:
I'm pretty sure all these contain statements can be simplified.

Reply:
<img width="311" alt="image" src="https://user-images.githubusercontent.com/9027725/234726745-a70e1159-018b-4a29-8fb7-afa43051f775.png">
<img width="242" alt="image" src="https://user-images.githubusercontent.com/9027725/234726752-5fe91811-5b88-4e02-b5e0-f9885828b2b2.png">

Reply:
+1. This can be simplified with the new `.?` and `??` operators.

Reply:
Interesting to know about this. lesser code. Used this at all places where ever possible.